internal/txapp/txapp.go

```diff
@@ -648,6 +678,47 @@ func (r *TxApp) Commit(ctx context.Context) error {
 
 	r.announceValidators()
 
+	// Take a snapshot of the database if node is not in the catchup mode and snapshots are enabled
+	if r.replayStatusFn != nil && !r.replayStatusFn() &&
+		r.snapshotter != nil && r.snapshotter.IsSnapshotDue(uint64(r.height)) {
```

Comment:
Try to order the conditions to exploit short-circuit behavior.  In particular, try to have `r.replayStatusFn()` behind other less expensive checks, especially if the other checks are likely to preclude the need to even check replay status almost all of the time.